baseplate/lint/db_query_string_format_plugin.py

```diff
@@ -0,0 +1,89 @@
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+class NoDbQueryStringFormatChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "no-database-query-string-format"
+    priority = -1
+    msgs = {
+        "W9000": (
+            "Python string formatting found in Cassandra database query. Database queries should be using Cassandra driver parameter substitution.",
+            "database-query-string-format",
+            "This allows CQL/SQL injection.",
+        )
+    }
+
+    def __init__(self, linter: PyLinter = None):
+        super().__init__(linter)
+        self.string_sub_queries: set = set()
+        self.query_verbs = {"select", "update", "delete", "insertinto", "truncate"}
+
+    def check_string_is_query(self, string: str) -> bool:
+        query = string.split(" ")
+        if query[0].lower() in self.query_verbs and ("{}" or "%s" in query):
+            return True
+        return False
+
+    def visit_assign(self, node: nodes) -> None:
+        """Check variables with queries using string formatting."""
+
+        variable_is_query = False
+        if (
+            isinstance(node.value, nodes.BinOp)
+            and node.value.op == "%"
+            and isinstance(node.value.left, nodes.Const)
+            and self.check_string_is_query(node.value.left.value)
+        ):
+            variable_is_query = True
+        elif (
+            isinstance(node.value, nodes.Call)
+            and isinstance(node.targets[0], nodes.Name)
+            and isinstance(node.value.func, nodes.Attribute)
+            and node.value.func.attrname == "format"
+            and self.check_string_is_query(node.value.func.expr.value)
+        ):
+            variable_is_query = True
+        if variable_is_query and node.targets[0].name not in self.string_sub_queries:
```

Comment:
readability thing: could you add a blank line before this `if` block? on first read, i thought these were more `elif` branches because they're all jammed together so i missed that `variable_is_query` gets used in these conditionals.